fdk/context.py

```diff
@@ -124,6 +150,87 @@ def __is_gateway(self):
                 and self.__headers.get(constants.FN_INTENT)
                 == constants.INTENT_HTTP_REQUEST)
 
+class TracingContext(object):
+    
+    def __init__( self, trace_collector_url, trace_id, 
+                  span_id, parent_span_id, 
+                  is_sampled, flags):
+      """
+      Tracing context here to be a placeholder
+      for tracing-specific attributes
+      :param trace_collector_url: APM Trace Collector Endpoint URL
+      :type trace_collector_url: str
+      :param trace_id: Trace ID
+      :type trace_id: str
+      :param span_id: Span ID
+      :type span_id: str
+      :param parent_span_id: Parent Span ID
+      :type parent_span_id: str
+      :param is_sampled: Boolean for emmitting spans
+      :type is_sampled: int (0 or 1)
+      :param flags: Debug flags
+      :type flags: int (0 or 1)
+      """  
+      self.__trace_collector_url = trace_collector_url
+      self.__trace_id = trace_id
+      self.__span_id = span_id
+      self.__parent_span_id = parent_span_id
+      self.__is_sampled = is_sampled
+      self.__flags = flags
+      self.__app_name = os.environ.get(constants.FN_APP_NAME)
+      self.__app_id = os.environ.get(constants.FN_APP_ID)
+      self.__fn_name = os.environ.get(constants.FN_NAME)
+      self.__fn_id = os.environ.get(constants.FN_ID)
+
+    def trace_collector_url(self):
+      return self.__trace_collector_url
+  
+    def trace_id(self):
+      return self.__trace_id
+
+    def span_id(self):
+      return self.__span_id
+
+    def parent_span_id(self):
+      return self.__parent_span_id
+
+    def is_sampled(self):
+      return bool(self.__is_sampled)
+
+    def flags(self):
+      return self.__flags
+
+    # this is a helper method specific for py_zipkin, this can be removed if it is too specific but as py_zipkin is the most popular I have left this in.
```

Comment:
Yea that is true, something to consider. I have talked about this with Sachin and we think its okay to leave it in. If we removed this the user who would most likely use py_zipkin as their choice of library would have to implement this themselves. This will also be present in the sample function code.